### ISSUE TITLE
[LiveComponent] Error: Could not update model for non HTMLElement

### ISSUE THREAD

**Issue by Lorenzschaef** [symfony/ux#650]
I've been building my first Live Component and I'm running into the following problem:

I have the following boolean live prop:

``` 
    #[LiveProp(writable: true)]
    public bool $overlayOpen = false;
```

In my view, I have the following button:
```    <button data-action="live#update" data-model="overlayOpen" data-value="1">Open overlay</button>```

When I clickt it, `$overlayOpen` is set to `true` and the component rerenders with a new element that contains the following button:

```<button data-action="live#update" data-model="overlayOpen" data-value="0">Close overlay</button>```

The "open" button works fine, but when clicking the "close" button, I get the following error:

```
Error invoking action "click->live#update"

 Error: Could not update model for non HTMLElement
    updateModelFromElementEvent live_controller.js:2260
    update live_controller.js:2198
    invokeWithEvent stimulus.js:281
    handleEvent stimulus.js:253
    handleEvent stimulus.js:31
```

The same happens in another instance, where I try to set a LiveProp to null.

I'm not sure if this is a bug or a problem on my side. Any help is appreciated.

----

I did some more testing.

I changed the property to a string containing "open" or "closed". The same error occurs. So it doesn't seem to relate to the fact that it is a "falsey" value. Rather I believe it has to do with the fact that these buttons appear only after the first rerender of the component. However, I have other buttons, that appear later on and do work correctly.

**Comment by 1ed** [comment]
Are the components embedded in each other?

**Comment by Lorenzschaef** [comment]
@1ed No, there is no nesting in this case.

**Comment by weaverryan** [comment]
Hey @Lorenzschaef!

This definitely smells like a bug to me. But I can't yet repeat it locally. My guess is that it has something to do with the elements hiding/showing, and so I set up something similar locally:

```
{% if isMessageShown %}
    I AM THE MESSAGE!!!
    <button data-action="live#update" data-model="isMessageShown" data-value="0">Hide message</button>
{% else %}
    <button data-action="live#update" data-model="isMessageShown" data-value="1">Show Message</button>
{% endif %}
```

But this seems to be working fine. Two things to check / debug:

A) Try going to the absolute, bleeding edge latest version of the bundle. I'm not aware of any specific fix that is unreleased, but I'm testing with the absolute latest code and that's where it's working fine. To do this, in `composer.json`, you'll need `"symfony/ux-live-component": "2.x-dev",` - then do a `yarn install --force` and `yarn watch`

B) Assuming the problem still happens, try hacking some debug code into the dist file. The error is that the "element" that was clicked to trigger the model change... somehow *isn't* an HTMLElement? I'm wondering what it is? In `node_modules/@symfony/ux-live-component/dist/live_controller.js`, find the line where the error is coming from - line `2260` in your stacktrace. It should look something like this:

```
if (!(element instanceof HTMLElement)) {
    throw new Error('Could not update model for non HTMLElement');
}
```

Right before the new Error, add:

```
console.log(element)
```

Then stop Encore and restart it so that it builds with this. What does that log?

Cheers!

**Comment by fGuix** [comment]
Hello @weaverryan 

I have the same issue and for me, the problem is apparently coming from the usage of svg icons (we use fontawesome kits).

In my code I have 2 links : 
```
// 1.
<a
        class="d-inline-flex link-layout-switcher{% if listDisplay is same as(false) %} active{% endif %}"
        data-model="listDisplay"
        data-value="0"
        data-action="live#update"
>
    <i class="fa-solid fa-grid fa-sm"></i>
</a>

// 2.
<a
        class="d-inline-flex link-layout-switcher{% if listDisplay is same as(true) %} active{% endif %}"
        data-model="listDisplay"
        data-value="1"
        data-action="live#update"
>
    <i class="fa-solid fa-list fa-sm"></i>
</a>
```

The first render is ok. But then clicking on the links gives the same error, with those console logs for the "non-HTMLElement" : 
```
// either
<svg class="svg-inline--fa fa-list fa-sm" aria-hidden="true" focusable="false" data-prefix="fas" data-icon="list" role="img" xmlns="http://www.w3.org/2000/svg" viewBox="0 0 512 512" data-fa-i2svg=""><path fill="currentColor" d="..."></path></svg>
// or
<path fill="currentColor" d="..."></path>
```

It looks like the different console logs are related to where we click on the icon (the whole svg or only path). 
I tried with inserting directly a super simple svg and still got the error.
Using only text instead of icons works fine.


I pushed the tests a bit further and seems like if we put html tags inside the link, live#update is triggered but the value is not updated (no ajax call).
What works : 
```
<a
        class="d-inline-flex link-layout-switcher{% if listDisplay is same as(true) %} active{% endif %}"
        data-model="listDisplay"
        data-value="1"
        data-action="live#update"
>
    List
</a>
```
what doesn't work (live#update logged in console but no ajax call)
``` 
<a
        class="d-inline-flex link-layout-switcher{% if listDisplay is same as(true) %} active{% endif %}"
        data-model="listDisplay"
        data-value="1"
        data-action="live#update"
>
    <p>List</p>
</a>
``` 
This might be another bug ? should it work ? Am I using it wrongly ?

Thanks in advance

**Comment by weaverryan** [comment]
Hey @fGuix!

Ok, thanks to you, we might be circling closer on the issue. Actually, there may be 2 separate things, and I need some more info about at least one of them.

About the SVG, I still don't quite understand your setup. You posted the HTML for 2 `<a>` tags that each have `data-action="live#update"`attached to them. That makes sense. But where does the SVG come into the picture? You also mentioned something about it depending on where you click, but I don't see any SVG inside those links? I'm missing a piece of the puzzle

Now, to the 2nd issue (I think this is unrelateed):

> I pushed the tests a bit further and seems like if we put html tags inside the link, live#update is triggered but the value is not updated (no ajax call).

So you said this this will NOT cause an Ajax call to trigger when the text is clicked:

```
<a
        class="d-inline-flex link-layout-switcher{% if listDisplay is same as(true) %} active{% endif %}"
        data-model="listDisplay"
        data-value="1"
        data-action="live#update"
>
    <p>List</p>
</a>
```

The cause might be simply that having a `<p>` (a block element) inside of `<a>` (an inline element, or `inline-flex` with the class you have on it) is not allowed... so your browser might even be doing something funny where the `<p>List</p>` is moved outside of the `<a>` when rendered. I could be totally wrong - but try it with a `<span>List</span>` and see if it's any better.

**Comment by Lorenzschaef** [comment]
A a matter of fact, I am using fontawesome too. I didn't include that in my example for simplicity. I didn't think of it as a possible origin of the problem, so sorry for that.

The svgs are inserted by a script, based on the classes in the `<i>` tag.

**Comment by fGuix** [comment]
Ah, yes, as mentioned by @Lorenzschaef , a fontawesome script is loaded in the head of the pages, and it changes the `<i>` tags with `<svg>` tags based on classes. I'm not particularly familiar with the underlying process, but on the rendered component, the icons are replaced correctly (and the result is the svg shown in my previous comment).
I tried with a simple svg like this : 
```
<a
        class="d-inline-flex link-layout-switcher{% if listDisplay is same as(true) %} active{% endif %}"
        data-model="listDisplay"
        data-value="1"
        data-action="live#update"
>
    List
    <svg height="210" width="500">
        <polygon points="100,10 40,198 190,78 10,78 160,198" style="fill:lime;stroke:purple;stroke-width:5;fill-rule:nonzero;"/>
    </svg>
</a>
```
When I click on the "List" part, everything works fine.
When I click on the `<svg>`, I got the "Could not update model for non HTMLElement" error.

---

For the 2nd issue, I tried with the `<span>` and the behavior is the same. I tried with this other test : 
```
<span>List</span>test
```
When I click on the "List" part, the `live #update` is logged but no ajax call triggered.
When I click on the "test" part, the `live #update` is logged and an ajax call is triggered (and the component is refreshed).
Should we (I) open another issue for this ?
